modules/vm/templates/entrypoint.sh.tpl

```diff
@@ -11,9 +11,95 @@ until ping -c 1 google.com &> /dev/null; do
   sleep 5
 done
 
-# TODO: Find/create/mount volumes
-# https://learn.microsoft.com/en-us/azure/virtual-machine-scale-sets/tutorial-use-disks-cli
+# Find/create/attach volumes
 
+instanceHostname=\'$(hostname)\'
+subscriptionID=$(az account show --query "id" --output tsv)
+resourceGroup=$(az vmss list --query "[0].resourceGroup" --output tsv)
+vmssName=$(az vmss list --query "[0].name" --output tsv)
+instanceID=$(az vmss list-instances --resource-group $resourceGroup --name $vmssName --query "[?contains(osProfile.computerName, ${instanceHostname})].instanceId" --output tsv)
```

Comment:
Use double `$$`, otherwise Terraform will try to evaulate it as a template variable.